### PR TITLE
Update unit tests' mock auth objects to use auto-speccing

### LINKED ISSUE #214: [ BUG ] Unit tests fail to run
## Describe the bug

Attempting to execute unit tests results in the following error:

```python3
    @property
    def token_status(self) -> int:
        """Provide the current token_status."""
>       return self.auth_object.token_status
E       AttributeError: 'CustomIOA' object has no attribute 'auth_object'

../../../Library/Caches/pypoetry/virtualenvs/caracara-ksX5R94U-py3.11/lib/python3.11/site-packages/falconpy/_service_class/_base_service_class.py:193: AttributeError
============================================================================================================= short test summary info ==============================================================================================================
FAILED tests/unit_tests/test_hosts.py::test_describe_devices - AttributeError: 'CustomIOA' object has no attribute 'auth_object'
FAILED tests/unit_tests/test_hosts.py::test_describe_devices__online_only - AttributeError: 'CustomIOA' object has no attribute 'auth_object'
[...]
```

This is because FalconPy checks that the `auth_object` given to a Service Class is an instance of `FalconInterface`
```python3
if auth_object:
    if isinstance(auth_object, FalconInterface):  # Issue 1043
```
and the current usage of `MagicMock` doesn't pass this check. The following patch fixes the problem
```diff
+++ b/tests/unit_tests/test_custom_ioas.py
@@ -25,7 +25,7 @@ def client():
     """Pytest fixture that provides a client with a mocked OAuth2 object, so no API credentials
     are needed.
     """
-    auth = MagicMock(autospec=falconpy.OAuth2)
+    auth = MagicMock(spec=falconpy.OAuth2, autospec=True)
     client = Client(falconpy_authobject=auth)
     return client
```
as the following snippet demonstrates
```python3
>>> auth = MagicMock(autospec=falconpy.OAuth2)
>>> isinstance(auth, falconpy.FalconInterface)
False
>>> auth = MagicMock(spec=falconpy.OAuth2, autospec=True)
>>> isinstance(auth, falconpy.FalconInterface)
True
```

## To Reproduce

`pytest tests/unit_tests`

## Expected behavior

The unit tests execute successfully

### Operating System Version

macOS Sonoma 14.6.1

### Python Version

Python 3.11.9

### Poetry Version

Poetry (version 1.8.4)

### Python Package Versions
<details>
<summary>Package versions collapsed, click to see</summary>

```
caracara-filters==1.0.0
certifi==2024.8.30
charset-normalizer==3.4.0
click==8.1.7
coverage==7.6.1
crowdstrike-falconpy==1.4.5
dill==0.3.9
flake8==7.1.1
idna==3.10
inflate64==1.0.0
iniconfig==2.0.0
isort==5.13.2
markdown-it-py==3.0.0
mccabe==0.7.0
mdurl==0.1.2
multivolumefile==0.2.3
mypy-extensions==1.0.0
packaging==24.1
pathspec==0.12.1
pbr==6.1.0
platformdirs==4.3.6
pluggy==1.5.0
prompt_toolkit==3.0.48
psutil==6.0.0
py7zr==0.22.0
pybcj==1.0.2
pycodestyle==2.12.1
pycryptodomex==3.21.0
pydocstyle==6.3.0
pyflakes==3.2.0
Pygments==2.18.0
pylint==3.2.7
pyppmd==1.1.0
pytest==8.3.3
python-dateutil==2.9.0.post0
PyYAML==6.0.2
pyzstd==0.16.2
requests==2.32.3
rich==13.9.2
six==1.16.0
snowballstemmer==2.2.0
stevedore==5.3.0
tabulate==0.9.0
texttable==1.7.0
toml==0.10.2
tomlkit==0.13.2
urllib3==2.2.3
wcwidth==0.2.13
```
</details>

## Additional context

Somewhat related: https://github.com/CrowdStrike/falconpy/issues/1043#issuecomment-1715731268

### PR DESCRIPTION
**Pull request by hur**
# Update unit tests' mock auth objects to use auto-speccing

- [ ] Enhancement
- [ ] Major feature update
- [X] Bug fixes
- [ ] Breaking change
- [ ] Updated unit tests
- [ ] Documentation

> Check the values above that match your PR and remove the remaining.

## Bandit analysis

```shell
(caracara-py3.11) USER-REDACTED@HOSTNAME-REDACTED tests % bandit -r . --configfile=.bandit
[main]	INFO	Found project level .bandit file: ./.bandit
[utils]	WARNING	Unable to parse config file ./.bandit or missing [bandit] section
[main]	INFO	profile include tests: None
[main]	INFO	profile exclude tests: B101
[main]	INFO	cli include tests: None
[main]	INFO	cli exclude tests: None
[main]	INFO	using config: .bandit
[main]	INFO	running on Python 3.11.9
Run started:2024-10-17 14:55:08.937788

Test results:
	No issues identified.

Code scanned:
	Total lines of code: 1445
	Total lines skipped (#nosec): 0

Run metrics:
	Total issues (by severity):
		Undefined: 0
		Low: 0
		Medium: 0
		High: 0
	Total issues (by confidence):
		Undefined: 0
		Low: 0
		Medium: 0
		High: 0
Files skipped (0):
```

## Added features and functionality

Fixed an issue with executing unit tests

## Issues resolved

- Fixes #214 by configuring auto-speccing for unit tests

### CODE REVIEW

**Comment by hur** [comment]
Note: 9 unit tests won't pass, these should probably be fixed in a separate issue / PR ? 

```shell
FAILED tests/unit_tests/test_custom_ioas.py::test_describe_rule_groups_with_rules - AssertionError: assert {'comment': '...000000Z', ...} == {'comment': '...000000Z', ...}
FAILED tests/unit_tests/test_custom_ioas.py::test_update_rule_groups_with_rule_changes - assert 1 == (1 + 1)
FAILED tests/unit_tests/test_hosts.py::test_contain - caracara.common.exceptions.MustProvideFilter: [500] You must provide a Falcon Filter in order to use this method
FAILED tests/unit_tests/test_hosts.py::test_release - caracara.common.exceptions.MustProvideFilter: [500] You must provide a Falcon Filter in order to use this method
FAILED tests/unit_tests/test_hosts.py::test_hide - caracara.common.exceptions.MustProvideFilter: [500] You must provide a Falcon Filter in order to use this method
FAILED tests/unit_tests/test_hosts.py::test_unhide - caracara.common.exceptions.MustProvideFilter: [500] You must provide a Falcon Filter in order to use this method
FAILED tests/unit_tests/test_hosts.py::test_tag - caracara.common.exceptions.MustProvideFilter: [500] You must provide a Falcon Filter in order to use this method
FAILED tests/unit_tests/test_hosts.py::test_untag - caracara.common.exceptions.MustProvideFilter: [500] You must provide a Falcon Filter in order to use this method
FAILED tests/unit_tests/test_prevention_policies.py::test_push_policy - TypeError: 'NoneType' object is not iterable
```

**Review by ChristopherHammond13 — APPROVED** [review]
